tests/source/bitbucket_server_test.py

```diff
@@ -34,10 +34,15 @@ def repos_project_response(mock_urlopen):
 
 
 @pytest.mark.usefixtures('repos_response')
-def test_list_repos():
-    settings = Settings(
-        'cool_user', 'app_password', 'bitbucket.domain.com',
-    )
+@pytest.mark.parametrize(
+    ('setting'),
+    (
+        ('cool_user', 'app_password', 'bitbucket.domain.com'),
+        ('cool_user', 'app_password', 'bitbucket.domain.com', None, True),
+    ),
+)
+def test_list_repos(setting):
```

Comment:
just make the `Settings` object in the parametrize itself rather than having indirection

also this doesn't demonstrate that your change works at all, it just lazily covers the code

Reply:
Used Settings object directly. I was looking for tests for request validation the repo and I could not find any. Am I missing something?